### ISSUE TITLE
How can this be used with a username/password in the env file?

### ISSUE THREAD

**Issue by jnovak-SM2Dev** [codemask-labs/nestjs-elasticsearch#202]
How can this be used with a username/password in the env file? In the `@nestjs/elasticsearch` module you would use `registerAsync` and use the `configService` like below. Is there a way to do the same thing with this package without having the auth info in the repo?

```
ElasticsearchModule.registerAsync({
      imports: [ConfigModule],
      useFactory: async (configService: ConfigService) => ({
        node: configService.get('ELASTICSEARCH_NODE'),
        auth: {
          username: configService.get('ELASTICSEARCH_USERNAME') ?? '',
          password: configService.get('ELASTICSEARCH_PASSWORD') ?? '',
        },
      }),
      inject: [ConfigService],
    }),
```

**Comment by przemyslawwalczak** [comment]
What do you mean by `auth info` in the repo?

We don't store `.env` on production environment like Kubernetes or Docker. Not sure how your application reads these environment variables but they are usually available through `process.env` (and coming from secrets) so they are safely stored 🤔 .

**Comment by przemyslawwalczak** [comment]
Unless, you are asking about `registerAsync` not available in the module?

**Comment by jnovak-SM2Dev** [comment]
I'm wondering how you pull in the env variables without access to `registerAsync` or `configService`.

**Comment by przemyslawwalczak** [comment]
yeah we should add `registerAsync` here 🤔 cc @ada-krupa @adrianboratyn

**Comment by adrianboratyn** [comment]
Hi @jnovak-SM2Dev we added registerAsync() method in the latest release https://github.com/codemask-labs/nestjs-elasticsearch/releases/tag/v2.10.0
